qemu/tests/disk_extension.py

```diff
@@ -0,0 +1,136 @@
+        process.system_output(cmd, shell=True)
+        device_img.create(device_img.params)
+        logging.debug("Finish creating loop device image {} on {}".format(loop_device_name, backend_filename))
+
+    def update_loop_device_backend_size(backend_img, device_img, size, unit="M"):
+        cmd = "qemu-img resize -f raw {} {}".format(backend_img.image_filename, str(size) + unit)
+        process.system_output(cmd, shell=True)
+        # update loop device
+        cmd = "losetup -c {0}".format(device_img.image_filename)
+        process.system_output(cmd, shell=True)
+
+    error_context.context("Get required parameter from the config file",
+                          logging.info)
+    current_size = int(params["begin_size"])
+    max_size = int(params["max_size"])
+    increment_size = int(params["increment_size"])
+    guest_cmd = params["guest_cmd"]
+    loop_device_backend_img_param = params.object_params('stg0')
+    loop_device_img_param = params.object_params('stg1')
+    drive_format = loop_device_img_param["drive_format"]
+    image_aio = loop_device_img_param["image_aio"]
+    if loop_device_backend_img_param["image_format"] != "raw":
+        test.error("Wrong loop device backend image format in config file.")
+
+    logging.debug("current_size:{}, max_size:{}".format(current_size, max_size))
+    logging.debug("increment_size:{}".format(increment_size))
+    logging.debug("guest_cmd: %s" % guest_cmd)
+    logging.debug("data disk drive_format:{} image_aio:{}".format(drive_format, image_aio))
+
+    error_context.context("Start to setup tmpfs folder", logging.info)
+    prepare_tmpfs_folder(params.get("tmpfs_folder", "/tmp/xtmpfs"))
+
+    error_context.context("Start to setup loop device", logging.info)
+    loop_device_backend_img = QemuImg(loop_device_backend_img_param, data_dir.get_data_dir(), 'stg0')
+    loop_device_img = QemuImg(loop_device_img_param, data_dir.get_data_dir(), 'stg1')
+    setup_loop_device_on_image(loop_device_backend_img, loop_device_img)
+
+    # start to boot vm
+    params["start_vm"] = "yes"
+    env_process.preprocess_vm(test, params, env, params["main_vm"])
+    error_context.context("Get the main VM", logging.info)
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+
+    timeout = int(params.get("login_timeout", 360))
+    session = vm.wait_for_login(timeout=timeout)
+
+    vm.verify_status("running")
+
+    disk_serial = params.get("disk_serial", "not_define")
+    disk_path = get_linux_drive_path(session, disk_serial)
+    if not disk_path:
+        test.fail("Can not find %s in guest." % disk_serial)
+
+    logging.debug("Find disk path %s for %s" % (disk_path, disk_serial))
+    guest_cmd = guest_cmd % disk_path
+
+    sleep_time = int(params.get("sleep_time", 3))
+    logging.debug("Sleep time is '%d' seconds", sleep_time)
+    time.sleep(sleep_time)
+
+    guest_cmd_output = session.sendline(guest_cmd)
+    logging.info(guest_cmd_output)
+
+    while current_size < max_size:
```

Comment:
I believe this whole space extension process should be included in a `try except` block to `fail` the case in case of certain exceptions, like:
```python
try:
    while current_size < max_size:
        pass
    except virt_vm.VMStatusError as e:
        test.fail("Got unexpected VM status %s" % repr(e)) 
```